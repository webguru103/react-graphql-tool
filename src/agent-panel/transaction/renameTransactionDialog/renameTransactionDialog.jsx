import React, { Component, Fragment } from 'react';
import DialogButtonBottom from '../../../shared/dialogButtonBottom/dialogButtonBottom';
import { withRenameTransaction } from '../api/transaction.service';
import RenameForm from './renameTransactionForm';
import { compose, pro } from '../../../utility';
import c from '../constants';
import type { TransactionType } from '../flowTypes';
import { withDialog } from '../../../shared/dialog/withDialog';

type Props = {
  transaction?: TransactionType,
  renameTransaction: Function,
  closeDialog: Function,
}

export class RenameTransactionDialogC extends Component<Props, *> {

  static defaultProps = {
    transaction: {
      id: '',
      name: '',
    },
  };

  state = {
    networkError: '',
  };

  rename = async ({ name }: { name: string } = {}) => {
    const { transaction, closeDialog, renameTransaction } = this.props;

    const [err] = await pro(renameTransaction({ newName: name, id: transaction ? transaction.id : '' }));

    if (err) {
      this.setState({
        networkError: c.ERROR_RENAMING_TRANSACTION,
      });
      return;
    }

    this.setState({
      networkError: '',
    });

    closeDialog();
  };

  render() {
    const { transaction } = this.props;
    return (
      <DialogButtonBottom
        title={(
          <Fragment>
            {c.RENAME_ITEM_DIALOG_HEADER}
            {` ${transaction ? transaction.name : ''}`}
          </Fragment>
        )}
        validate={(values) => {
          const errors = {};
          if (!values.name) {
            errors.name = c.ADD_NEW_NAME;
          }

          return errors;
        }}
        initialValues={{ name: (transaction && transaction.name) || '' }}
        content={<RenameForm networkError={this.state.networkError} />}
        actionButtonText={c.RENAME_DIALOG_SUBMIT}
        onSubmit={this.rename}
      />
    );
  }

}

export default compose(
  withRenameTransaction,
  withDialog,
)(RenameTransactionDialogC);
