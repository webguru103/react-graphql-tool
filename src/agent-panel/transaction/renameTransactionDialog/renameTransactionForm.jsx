import * as React from 'react';
import classNames from 'classnames';
import TextField from '@material-ui/core/TextField';
import { withStyles } from '@material-ui/core/styles';
import styles from './renameTransactionForm.styles';
import { noop } from '../../../utility';

type Props = {
  classes: Object,
  values?: Object,
  errors?: {name: string},
  handleChange?: Function,
  networkError: React.Node
}

const RenameTransactionC = ({
  classes, values, errors, handleChange, networkError,
}: Props) => (
  <TextField
    error={Boolean((errors && errors.name) || networkError)}
    fullWidth
    helperText={(errors && errors.name) || networkError}
    InputProps={{
      classes: {
        root: classes.textFieldInputWrapper,
        input: classes.textFieldInput,
        error: classes.textFieldInputError,
      },
      disableUnderline: true,
    }}
    InputLabelProps={{
      classes: { root: `${classNames(classes.textFieldLabel, classes.label)}` },
    }}
    onChange={handleChange}
    value={values && values.name}
    name="name"
  />
);

RenameTransactionC.defaultProps = {
  errors: {
    name: '',
  },
  values: {
    name: '',
  },
  handleChange: noop,
};

export default withStyles(styles, { withTheme: true })(RenameTransactionC);
