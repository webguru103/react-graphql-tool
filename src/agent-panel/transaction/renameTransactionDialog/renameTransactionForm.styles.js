const styles = (theme: Object) => ({
  toggle: {
    marginTop: 20,
    color: `${theme.palette.neutrals.secondaryText}`,
    '& p': {
      margin: 0,
    },
  },
  textFieldInputWrapper: {
    'label + &': {
      marginTop: '10px',
    },
  },
  textFieldInput: {
    padding: '10px 12px',
    border: `1px solid ${theme.palette.neutrals.greyUI}`,
    borderRadius: 4,
    color: `${theme.palette.neutrals.secondaryText}`,
    fontSize: 14,
    '&:focus': {
      borderColor: `${theme.palette.neutrals.secondaryText}`,
    },
  },
  textFieldInputError: {
    '& $textFieldInput': {
      borderColor: '#f44336',
    },
  },
  textFieldLabel: {
    position: 'relative',
    transform: 'none',
    fontSize: 14,
  },
  label: {
    color: `${theme.palette.neutrals.secondaryText}`,
    '& span': {
      color: `${theme.palette.neutrals.secondaryText}`,
    },
  },
});

export default styles;
