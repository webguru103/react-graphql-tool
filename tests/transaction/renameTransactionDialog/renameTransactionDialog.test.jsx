import React from 'react';
import { RenameTransactionDialogC } from '../../../src/agent-panel/transaction/renameTransactionDialog/renameTransactionDialog';
import { shallowWrap, mountWrap } from '../../test-helpers/contextWrapper';

describe('Rename', () => {
  let props;
  let component;
  let renameFunction = jest.fn();
  const wrappedShallow = (lang = 'en') => {
    component = shallowWrap(<RenameTransactionDialogC {...props} />, lang);
    return component;
  };

  const wrappedMount = (lang = 'en') => {
    component = mountWrap(<RenameTransactionDialogC {...props} />, lang);
    return component;
  };

  beforeEach(() => {
    renameFunction = jest.fn(({ transactionId, newName }) => Promise.resolve({ transactionId, newName }));
    props = {
      renameTransaction: renameFunction,
      transaction: {
        id: 'mockId',
        name: 'Transaction name',
      },
      closeDialog: jest.fn(),
    };
    if (component) component.unmount();
  });

  test('should render', () => {
    const wrapper = wrappedShallow();
    expect(wrapper).toMatchSnapshot();
  });

  test('should call renameTransaction with correct argument', (done) => {
    const wrapper = wrappedMount();

    // Should not have made a call yet
    expect(renameFunction.mock.calls.length).toBe(0);

    // Set the input value
    wrapper.find('input').simulate('change', { target: { name: 'name', value: 'Transaction name' } });
    // Click Done button (Dialog is set to open)
    wrapper.find('button').findWhere(n => n.contains('Done')).first().simulate('submit');

    // Calls renameTransaction, which returns a function that should be called once
    // with proper variables passed in, wrapped in a Promise.
    setTimeout(() => {
      expect(renameFunction).toHaveBeenCalledWith({ id: props.transaction.id, newName: props.transaction.name });
      done();
    });
  });

});
