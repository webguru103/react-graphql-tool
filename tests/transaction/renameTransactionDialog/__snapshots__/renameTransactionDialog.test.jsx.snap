// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rename should render 1`] = `
<WithStyles(DialogCreatorInjector)
  actionButtonText={
    <FormattedMessage
      defaultMessage="Done"
      id="rename-dialog-submit"
      values={Object {}}
    />
  }
  content={
    <WithStyles(RenameTransactionC)
      networkError=""
    />
  }
  initialValues={
    Object {
      "name": "Transaction name",
    }
  }
  onSubmit={[Function]}
  title={
    <React.Fragment>
      <FormattedMessage
        defaultMessage="Rename Item:"
        id="rename-item-dialog-header"
        values={Object {}}
      />
       Transaction name
    </React.Fragment>
  }
  validate={[Function]}
/>
`;
